doc/source/advanced/optimizers.rst

```diff
@@ -0,0 +1,95 @@
+Using the Exact Geodesic Transport with Conjugate Gradients(EGT-CG) Optimizer
+-------------------------------------------------------------------------------
+
+The Exact Geodesic Transport with Conjugate Gradients (EGT-CG) optimizer is a curvature-aware Riemannian optimizer designed specifically for variational circuits based on the Hamming-weight encoder (HWE) ansatze.
+
+It updates parameters along exact geodesic paths on the hyperspherical manifold defined by the HWE, combining analytic metric computation, conjugate-gradient memory, and dynamic learning rates for fast, globally convergent optimization.
+
+
+For more details, see:
+
+A. J. Ferreira‑Martins et al., *Variational quantum algorithms with exact geodesic transport*, 
+
+`arXiv:2506.17395 (2025) <https://arxiv.org/abs/2506.17395>`_.
+
+
+The current configuration focuses on ground state estimation problems where given a variational state :math:`\ket{\psi(\boldsymbol{\theta})}` parameterized by hyperspherical angles :math:`\boldsymbol{\theta}`, 
+
+the loss function minimized is the energy expectation value of the given hamiltonian:
+
+.. math::
+    \mathcal{L}(\boldsymbol{\theta}) = \bra{\psi(\boldsymbol{\theta})} \, H \, \ket{\psi(\boldsymbol{\theta})} \, ,
+
+But it is important to note that this framework generalizes to arbitrary loss functions.
+
+
+Defining and running the optimizer
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+In `qiboml`, the EGT-CG optimizer is implemented in the class
+:class:`qiboml.models.optimizers.ExactGeodesicTransportCG`.
+
+.. autoclass:: qiboml.models.optimizers.ExactGeodesicTransportCG
+   :members:
+   :undoc-members:
+   :show-inheritance:
+
+It requires as inputs the number of qubits, the Hamming weight, a Hamiltonian, and your initial parameters.
+
+**Example usage:**
+
+.. code-block:: python
+
+    import numpy as np
+    from qibo import hamiltonians
+    from scipy.special import comb
+    from qiboml.models.optimizers import ExactGeodesicTransportCG
+
+    nqubits = 4
+    weight = 2
+    dim = int(comb(nqubits, weight))
+    np.random.seed(42)
+
+    # Initial angles in hyperspherical coordinates
+    theta_init = np.random.uniform(low=0, high=np.pi, size=dim - 1)
+
+    # Define the Hamiltonian
+    hamiltonian = hamiltonians.XXZ(nqubits=nqubits)
```

Comment:
```suggestion
    theta_init = np.random.uniform(low=0, high=np.pi, size=dim - 1)

    # Define the Hamiltonian
    hamiltonian = XXZ(nqubits=nqubits)
```

I don't remember if the optimizer is supporting this, but the angles should be something other than `np.ndarray` objects (e.g. `torch.tensor`) if one is interested in vanilla backpropagation without using our custom way of calculating gradients.